contracts/Router.sol

```diff
@@ -84,11 +85,9 @@ contract Router {
 
         totalPledged = totalPledged + _amount;
 
-        uint256 pledgerId = getPledgerId(msg.sender);
-        if (pledgerId == 0) {
-            // user has not pledged before
-            pledgerId = numPledgers++;
-            pledgers[pledgerId] = msg.sender;
+        if (amountsPledged[msg.sender] == 0) {
+            // User has not pledged before. Add them to pledgers[] so we can loop over them.
+            pledgers[++numPledgers] = msg.sender;
```

Comment:
Yes, they could bloat the mapping that way, but for the same price in gas they could also bloat the mapping by creating many ethereum addresses with a tiny amount of dai in each and pledging with them.  I'm not sure how big the mapping would have to be to break everything, but it could also happen if there are a ton of legitimate users.  Rebase isn't doing any transfers--just math, but we should see if we can figure out how many pledgers or burners it would take to break that function.

Reply:
Actually, now that I look more carefully at what's inside those loops, if someone tried to attack the contract by repeatedly pledging and unpledging (or pledging with many accounts), they would like pay gas that would then be refunded to whoever calls rebase, because the main operations inside the loops are deletes. The attack would backfire and lose money for the attacker and give a gas refund to the rebase caller.